pbts does not properly handle filepaths with special characters in it

I'm contributing to another project that uses pbts. When I run `yarn run --silent pbts proto.js` from my project directory ``~/j"oker `something` 'cool  streamlit;`` that I made to test resiliency to special characters, I get this output:

```
Error: Command failed: yarn run --silent pbts proto.js
/bin/sh: 1: Syntax error: Unterminated quoted string
/home/user/j"oker `something` 'cool  streamlit;/frontend/node_modules/protobufjs-cli/pbts.js:129
                throw err;
                ^

Error: code 2
    at ChildProcess.<anonymous> (/home/user/j"oker `something` 'cool  streamlit;/frontend/node_modules/protobufjs-cli/pbts.js:126:27)
    at ChildProcess.emit (node:events:519:35)
    at maybeClose (node:internal/child_process:1101:16)
    at Socket.<anonymous> (node:internal/child_process:457:11)
    at Socket.emit (node:events:507:28)
    at Pipe.<anonymous> (node:net:346:12)

Node.js v24.2.0

Failed to generate protobuf files: Error: Command failed: yarn run --silent pbts proto.js
/bin/sh: 1: Syntax error: Unterminated quoted string
/home/user/j"oker `something` 'cool  streamlit;/frontend/node_modules/protobufjs-cli/pbts.js:129
                throw err;
                ^

Error: code 2
    at ChildProcess.<anonymous> (/home/user/j"oker `something` 'cool  streamlit;/frontend/node_modules/protobufjs-cli/pbts.js:126:27)
    at ChildProcess.emit (node:events:519:35)
    at maybeClose (node:internal/child_process:1101:16)
    at Socket.<anonymous> (node:internal/child_process:457:11)
    at Socket.emit (node:events:507:28)
    at Pipe.<anonymous> (node:net:346:12)

Node.js v24.2.0

    at genericNodeError (node:internal/errors:983:15)
    at wrappedFn (node:internal/errors:537:14)
    at ChildProcess.exithandler (node:child_process:415:12)
    at ChildProcess.emit (node:events:507:28)
    at maybeClose (node:internal/child_process:1101:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5) {
  code: 1,
  killed: false,
  signal: null,
  cmd: 'yarn run --silent pbts proto.js'
}
make: *** [Makefile:237: protobuf] Error 1
```

I'm happy to tell you I know exactly why this happens, after just a little sleuthing: https://github.com/protobufjs/protobuf.js/blob/f42297b29d15c8e0382744a83f5147a1aa978f42/cli/pbts.js#L97

Specifically, this call to exec does not escape its arguments (note that escaping arguments is generally a pretty brittle and annoying way of dealing with such problems...), so they evaluate in the shell and cause a parse error:

```
 function callJsdoc() {

        // There is no proper API for jsdoc, so this executes the CLI and pipes the output
        var basedir = path.join(__dirname, ".");
        var moduleName = argv.name || "null";
        var nodePath = process.execPath;
        var cmd = "\"" + nodePath + "\" \"" + require.resolve("jsdoc/jsdoc.js") + "\" -c \"" + path.join(basedir, "lib", "tsd-jsdoc.json") + "\" -q \"module=" + encodeURIComponent(moduleName) + "&comments=" + Boolean(argv.comments) + "\" " + files.map(function(file) { return "\"" + file + "\""; }).join(" ");
        var child = child_process.exec(cmd, {
            cwd: process.cwd(),
            argv0: "node",
            stdio: "pipe",
            maxBuffer: 1 << 26 // 67mb
        });
```

Incidentally, we also use `yarn run --silent pbjs`, and that has no problem.

--- a/cli/pbts.js
+++ b/cli/pbts.js
@@ -98,12 +98,17 @@ exports.main = function(args, callback) {
         var basedir = path.join(__dirname, ".");
         var moduleName = argv.name || "null";
         var nodePath = process.execPath;
-        var cmd = "\"" + nodePath + "\" \"" + require.resolve("jsdoc/jsdoc.js") + "\" -c \"" + path.join(basedir, "lib", "tsd-jsdoc.json") + "\" -q \"module=" + encodeURIComponent(moduleName) + "&comments=" + Boolean(argv.comments) + "\" " + files.map(function(file) { return "\"" + file + "\""; }).join(" ");
-        var child = child_process.exec(cmd, {
+        var jsdocArgs = [
+            require.resolve("jsdoc/jsdoc.js"),
+            "-c",
+            path.join(basedir, "lib", "tsd-jsdoc.json"),
+            "-q",
+            "module=" + encodeURIComponent(moduleName) + "&comments=" + Boolean(argv.comments)
+        ].concat(files);
+        var child = child_process.spawn(nodePath, jsdocArgs, {
             cwd: process.cwd(),
             argv0: "node",
-            stdio: "pipe",
-            maxBuffer: 1 << 26 // 67mb
+            stdio: "pipe"
         });
         var out = [];
         var ended = false;

--- a/tests/cli.js
+++ b/tests/cli.js
@@ -5,6 +5,8 @@ var path = require("path");
 var Module = require("module");
 var protobuf = require("..");
 var fs = require("fs");
+var EventEmitter = require("events").EventEmitter;
+var child_process = require("child_process");
 
 function cliTest(test, testFunc) {
     // pbjs does not seem to work with Node v4, so skip this test if we're running on it
@@ -107,6 +109,38 @@ tape.test("pbjs generates correct ES6 static-module imports", function(test) {
             test.ok(jsCode.includes("import $protobuf from \"protobufjs/minimal.js\";"), "es6 wrapper uses a default import and explicit .js extension");
             test.end();
         });
+    });
+});
+
+tape.test("pbts passes jsdoc arguments without a shell", function(test) {
+    var pbts = require("../cli/pbts");
+    var originalSpawn = child_process.spawn;
+    var file = "j\"oker `something` 'cool  streamlit;.js";
+
+    test.plan(5);
+
+    child_process.spawn = function(cmd, args, options) {
+        var child = new EventEmitter();
+        child.stdout = new EventEmitter();
+        child.stderr = { pipe: function() {} };
+
+        test.equal(cmd, process.execPath, "should execute node directly");
+        test.ok(/jsdoc[\\/]jsdoc\.js$/.test(args[0]), "should execute jsdoc directly");
+        test.equal(args[args.length - 1], file, "should pass file path as a single argument");
+        test.equal(options.stdio, "pipe", "should pipe jsdoc output");
+
+        process.nextTick(function() {
+            child.stdout.emit("data", "declare namespace test {}\n");
+            child.stdout.emit("end");
+            child.emit("close", 0);
+        });
+
+        return child;
+    };
+
+    pbts.main([file], function(err) {
+        child_process.spawn = originalSpawn;
+        test.error(err, "should generate definitions");
     });
 });
 

--- a/tests/cli.js
+++ b/tests/cli.js
@@ -115,7 +115,7 @@ tape.test("pbjs generates correct ES6 static-module imports", function(test) {
 tape.test("pbts passes jsdoc arguments without a shell", function(test) {
     var pbts = require("../cli/pbts");
     var originalSpawn = child_process.spawn;
-    var file = "j\"oker `something` 'cool  streamlit;.js";
+    var file = "file with \"quotes\" `backticks` 'apostrophes' and ;.js";
 
     test.plan(5);
 

